templates/habootstrap-formula/cluster_resources_nfs.j2

```diff
@@ -21,36 +21,49 @@ clone cl-nfsserver nfsserver \
 
 # Need to edit drbd pillar file.
 {%- for res in drbd.resource %}
-primitive drbd-{{ res.name }} ocf:linbit:drbd \
+primitive rsc_drbd-{{ res.name }} ocf:linbit:drbd \
   params drbd_resource={{ res.name }} \
   op monitor interval=15 role=Master \
   op monitor interval=30 role=Slave
 
-ms ms_{{ res.name }} drbd-{{ res.name }} \
+ms ms_{{ res.name }} rsc_drbd-{{ res.name }} \
   meta master-max=1 master-node-max=1 \
   meta clone-max={{ res.nodes | length }} clone-node-max=1 \
   meta notify=true target-role=Started
 
-primitive fs_{{ res.name }} ocf:heartbeat:Filesystem \
+primitive rsc_fs_{{ res.name }} ocf:heartbeat:Filesystem \
   params device={{ res.device }} directory={{ res.mount_point }} fstype={{ res.file_system }} \
   options=noatime,nodiratime \
   op monitor interval="20" timeout="40s"
 
-primitive test-IP_{{ res.name }} IPaddr2 \
-        params ip={{ res.virtual_ip }}
-
-primitive exportfs_work_{{ res.name }} exportfs \
+primitive rsc_exportfs_{{ res.name }} exportfs \
   params directory={{ res.mount_point }} fsid={{ nfsid + loop.index0 }} \
   options="rw,no_root_squash" clientspec="*" wait_for_leasetime_on_stop=true \
   op monitor interval=30s
 
-group g-nfs_{{ res.name }} fs_{{ res.name }} test-IP_{{ res.name }} exportfs_work_{{ res.name }}
+{%- if res.virtual_ip is defined %}
```

Comment:
Why do we need an additional IP resource?
I think the the IP address should come from `data.virtual_ip`

Reply:
This `res.virtual_ip` is per DRBD resource virtual ip address, not the same one (`data.virtual_ip`) for the node.
In case some users wants different NFS server have different virtual IP assigned.

I think the same name `virtual_ip` (`data.virtual_ip` and `drbd.resource.virtual_ip`) do confusing. If no customer use per resource vip, maybe we can remove this. Or change the name of `drbd.resource.virtual_ip` to `drbd.resource.res_vip`

Reply:
This `res.virtual_ip` is per DRBD resource virtual ip address, not the same one (`data.virtual_ip`) for the node.
In case some users wants different NFS server have different virtual IP assigned.

I think the same name `virtual_ip` (`data.virtual_ip` and `drbd.resource.virtual_ip`) do confusing. If no customer use per resource vip, maybe we can remove this. Or change the name of `drbd.resource.virtual_ip` to `drbd.resource.res_vip`

Reply:
This `res.virtual_ip` is per DRBD resource virtual ip address, not the same one (`data.virtual_ip`) for the node.
In case some users wants different NFS server have different virtual IP assigned.

I think the same name `virtual_ip` (`data.virtual_ip` and `drbd.resource.virtual_ip`) do confusing. If no customer use per resource vip, maybe we can remove this. Or change the name of `drbd.resource.virtual_ip` to `drbd.resource.res_vip`

Reply:
This is basically the implementation as it was before (without a cluster wide VIP). But yes, I also do not see the need to implement this at the moment.